speaker/layer2_controller.go

```diff
@@ -153,6 +155,20 @@ func (c *layer2Controller) SetNode(log.Logger, *v1.Node) error {
 	return nil
 }
 
+func ipAdvertisementFor(ip net.IP, localNode string, l2Advertisements []*config.L2Advertisement) layer2.IPAdvertisement {
+	ifs := sets.NewString()
+	for _, l2 := range l2Advertisements {
+		if matchNode, ok := l2.Nodes[localNode]; !ok || !matchNode {
+			continue
```

Comment:
I think we will never end up in that situation (because ShouldAnnounce would prevent it), but would it make sense to throw an error and log if there are no matching advertisements?

Reply:
Can you check this comment?